joeynmt/data.py

```diff
@@ -101,29 +101,56 @@ def load_data(data_cfg: dict) -> (Dataset, Dataset, Optional[Dataset],
     return train_data, dev_data, test_data, src_vocab, trg_vocab
 
 
-def make_data_iter(dataset: Dataset, batch_size: int, train: bool = False,
+# pylint: disable=global-at-module-level
+global max_src_in_batch, max_tgt_in_batch
```

Comment:
Why are global variables needed here? Can we avoid them?

Reply:
I'm not sure so I'd like to leave it for a future improvement. I also don't like them!